group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -4308,9 +4308,18 @@ public void replay(
         String groupId = key.groupId();
         String regex = key.regularExpression();
 
+        ConsumerGroup consumerGroup;
+        try {
+            consumerGroup = getOrMaybeCreatePersistedConsumerGroup(groupId, value != null);
+        } catch (GroupIdNotFoundException ex) {
+            // If the group does not exist and a tombstone is replayed, we can ignore it.
```

Comment:
This is a bit subtle. I suppose we will never throw the exception is `value != null`, so we know that a tombstone is replayed if we enter the `catch` block. Making this explicit in the comment would have helped me, but I'm fine with leaving it like that.